proto/tendermint/liquidity/v1beta1/tx.proto

```diff
@@ -0,0 +1,207 @@
+  option (gogoproto.goproto_getters) = false;
+
+  string withdrawer_address = 1 [ (gogoproto.moretags) = "yaml:\"withdrawer_address\"",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "account address of the origin of this message",
+      example: "\"cosmos1e35y69rhrt7y4yce5l5u73sjnxu0l33wvznyun\"",
+      format: "sdk.AccAddress"
+    }];
+  // id of the target pool
+  uint64 pool_id = 2 [(gogoproto.moretags) = "yaml:\"pool_id\"", (gogoproto.jsontag) = "pool_id",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"1\"",
+      format: "uint64"
+    }];
+  cosmos.base.v1beta1.Coin pool_coin = 3 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"pool_coin\"",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "{\"denom\": \"poolD35A0CC16EE598F90B044CE296A405BA9C381E38837599D96F2F70C2F02A23A4\", \"amount\": \"1000\"}",
+      format: "sdk.Coin"
+    }];
+}
+
+// MsgWithdrawWithinBatchResponse defines the Msg/WithdrawWithinBatch response type.
+message MsgWithdrawWithinBatchResponse {}
+
+// `MsgSwapWithinBatch` defines an sdk.Msg type that supports submitting swap offer request to the batch of the liquidity pool
+// Swap offer submit to the batch to the Liquidity pool with the specified the `pool_id`, `swap_type_id`,
+// `demand_coin_denom` with the coin and the price you're offering
+// and `offer_coin_fee` must half of offer coin amount * current `params.swap_fee_rate` for reservation to pay fees
+// this requests are stacked in the batch of the liquidity pool, not immediately processed and
+// processed in the `endblock` at once with other requests
+// You should request the same each field as the pool
+// Currently, only the default `swap_type_id`1 is available on this version
+// The detailed swap algorithm can be found here.
+//
+// See: https://github.com/tendermint/liquidity/tree/develop/doc
+// https://github.com/tendermint/liquidity/blob/develop/x/liquidity/spec/04_messages.md
+message MsgSwapWithinBatch {
+  option (gogoproto.equal)           = false;
+  option (gogoproto.goproto_getters) = false;
+  // address of swap requester
+  string swap_requester_address = 1 [(gogoproto.moretags) = "yaml:\"swap_requester_address\"",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      description: "account address of the origin of this message",
+      example: "\"cosmos1e35y69rhrt7y4yce5l5u73sjnxu0l33wvznyun\"",
+      format: "sdk.AccAddress"
+    }];
+  // id of the target pool
+  uint64 pool_id = 2 [(gogoproto.moretags) = "yaml:\"pool_id\"", (gogoproto.jsontag) = "pool_id",
+    (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {
+      example: "\"1\"",
+      format: "uint64"
+    }];
+
+  // id of swap type, only 1 is allowed on this version, Must match the value in the pool.
+  uint32 swap_type_id = 3 [(gogoproto.moretags) = "yaml:\"swap_type_id\"",
```

Comment:
hmmm is swap_type_id 3 supported? line 162 used to say only 1 is allowed (which I _think_ means that swap_type_id must be 1)

Reply:
That number 3 doesn't mean swap_type_id is 3. That is just a unique number for a field when you define message. [Check out this link to understand further](https://developers.google.com/protocol-buffers/docs/proto3#assigning_field_numbers).